src/hooks/useVortexAccount.ts

```diff
@@ -0,0 +1,56 @@
+import { Networks, useNetwork } from '../contexts/network';
+import { useMemo } from 'preact/compat';
+import { usePolkadotWalletState } from '../contexts/polkadotWallet';
+import { useAccount } from 'wagmi';
+
+// For the AssetHub network, we use a chain ID of -1. This is not a valid chain ID
+// but we just use it to differentiate between the EVM and Polkadot accounts.
+const AssetHubChainId = -1;
+
+// A helper hook to provide an abstraction over the account used.
+// The account could be an EVM account or a Polkadot account.
+export const useVortexAccount = () => {
+  const { selectedNetwork } = useNetwork();
+
+  const { walletAccount: polkadotWalletAccount } = usePolkadotWalletState();
+  const { isDisconnected: isEvmAccountDisconnected, chainId: evmChainId, address: evmAccountAddress } = useAccount();
+
+  const address = useMemo(() => {
+    if (selectedNetwork === Networks.AssetHub) {
+      return polkadotWalletAccount?.address;
+    } else {
+      return evmAccountAddress;
+    }
+  }, [evmAccountAddress, polkadotWalletAccount, selectedNetwork]);
+
+  const isDisconnected = useMemo(() => {
+    if (selectedNetwork === Networks.AssetHub) {
+      return !polkadotWalletAccount;
+    } else {
+      return isEvmAccountDisconnected;
+    }
+  }, [selectedNetwork, polkadotWalletAccount, isEvmAccountDisconnected]);
+
+  const chainId = useMemo(() => {
+    if (selectedNetwork === Networks.AssetHub) {
+      return AssetHubChainId;
+    } else {
+      return evmChainId;
+    }
+  }, [selectedNetwork, evmChainId]);
+
+  const type = useMemo(() => {
```

Comment:
I think we don't have to useMemo for variables here, there is no heavy computation that would require it, wdyt?

Reply:
That's true but I decided to use it anyway to mitigate any potential future issues in other components if we decide to put the return values of this hook into the dependencies of another hook. We had those issues in the past and it was quite hard to debug IMO.